learn/security/tenant_tokens.md

```diff
@@ -0,0 +1,299 @@
+ }
+}
+```
+
+The object key must be an index name. You may also use `*` instead of a specific index name—in this case, search rules will be applied to all indexes.
+
+The object value must consist of `search_parameter:value` pairs. Currently, **tenant tokens only supports the `filter` [search parameter](/reference/api/search.md#filter)**.
+
+The following search rule forces every query across all accessible indexes to only return results whose `user_id` equals `1`:
+
+```json
+{
+  "*": {
+    "filter": "user_id = 1"
+  }
+}
+```
+
+The next rule goes a bit further. When searching on the `patient_medical_records` index, a user can only see records that belong to them and have been marked as published:
+
+```json
+{
+  "patient_medical_records": {
+    "filter": "user_id = 1 AND published = true"
+  }
+}
+```
+
+A token may contain rules for any number of indexes. **Specific rulesets take precedence and overwrite wildcard rules.**
+
+The previous rules can be combined in one tenant token:
+
+```json
+{
+  "apiKeyPrefix": "rkDxFUHd",
+  "exp": 1641835850,
+  "searchRules": {
+    "*": {
+      "filter": "user_id = 1"
+    },
+    "medical_records": {
+      "filter": "user_id = 1 AND published = true",
+    }
+  }
+}
+```
+
+In [the previous SDK example](#generating-tenant-tokens-with-an-sdk), the search rules are stored in `searchRules`:
+
+```js
+// placeholder code sample. Final version will be supplied by the integrations team
+// <CodeSamples id="tenant_token_guide_fragment_rules_1">
+const searchRules = {
+  'patient_medical_records': {
+    'filter': `user_id = ${currentUserID}`
+  }
+};
+const token = client.generateTenantToken(searchRules, expiryDate, apiKey);
+```
+
+::: danger
+Because tenant tokens are generated in your application, Meilisearch cannot check if search rule filters are valid. Invalid search rules will only throw errors when they are used in a query.
+
+Consult the search API reference [more information on Meilisearch filter syntax](/reference/api/search.md#filter).
+:::
+
+### API Keys
+
+Creating a token requires an API key. A token has access to the same indexes as the API key used to generate it.
+
+Since a master key is not an API key, **you cannot use a master key to create a tenant token**.
+
+For security reasons, we strongly recommend you avoid exposing the API key whenever possible and **always generate tokens on your application's back-end**.
+
+When using an official Meilisearch SDK, you may indicate which API key you wish to use when generating a token. Consult the documentation of the SDK you are using for more specific instructions.
+
+In [the previous SDK example](#generating-tenant-tokens-with-an-sdk), the API key is stored in `apiKey`:
+
+```js
+// placeholder code sample. Final version will be supplied by the integrations team
+// <CodeSamples id="tenant_token_guide_fragment_key_1">
+const apiKey = '[an_api_key]';
+const token = client.generateTenantToken(searchRules, expiryDate, apiKey);
```

Comment:
```suggestion
const token = client.generateTenantToken(searchRules, { apiKey });
```

ApiKey is actually optional because the apiKey provided by the client `client` can also be used. 
This works: 

```js
const client = new MeiliSearch({ host: "...", apiKey: "1234567" })

const token = client.generateToken(["*])
```
In this case it will use the `apiKey` of the `client`